### PR TITLE
Add magidoc documentation

### LINKED ISSUE #45: Proposition: add static documentation
Hello! I am the maintainer of a documentation tool called [Magidoc](https://github.com/magidoc-org/magidoc), a tool that generates static GraphQL API documentation references. 

I would like to propose to integrate Magidoc into your great project to build a static API reference. This would allow users to easily get an overview of the whole API, but also other documentation that may be useful to use your API. You could even move all your examples or any other content you want there, since Magidoc supports `Custom pages`.

If you want to see an example, here is a link to a [demo website](https://magidoc-carbon-multi-page.netlify.app/welcome).

I am also willing to help you integrate it if you want, and this could be hosted on any static content host: so can be GitHub pages, Netlify, Vercel, or anything else you want.

Let me know what you think about the idea, and feel free to close the issue if you're not interested.

### PR DESCRIPTION
**Pull request by pelletier197**
Added magidoc documentation. See the example output here: https://pelletier197.github.io/countries

Once merged, the documentation will automatically be deployed at https://trevorblades.github.io/countries

To simplify my life, I needed to extract the schema in a separate file. This is easier to get the schema for me, and will be simpler for you to add the comments.

- For now, custom pages are empty. Only the generated documentation is shown. Feel free to add markdown pages if you want and move your examples in there. (See: https://magidoc.js.org/cli/magidoc-configuration#website)
- Documentation is rebuilt and deployed at every merge on the main branch. 

Thank you!

Resolve #45

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import {ApolloServer} from 'apollo-server';
-import {resolvers, typeDefs} from './schema.js';
+import {resolvers, typeDefs} from './schema.mjs';
```
Comment (pelletier197) [inline index.js:2]: I don't understand why I had to do this change, but eslint was complaining. Guess it wasn't complaning before, but any idea how this worked before and stopped working with my PR?

**Comment by pelletier197** [comment]
@trevorblades any chance you can check this ?

File: index.js (hunk 1 above)
Comment (trevorblades) [reply index.js:2]: I wonder if it has anything to do with the version of node you're running. I'll try to run it on my machine and see if it's still unhappy.

**Comment by trevorblades** [comment]
I made a branch with a bunch of changes that make the linter happy again. Could you add me as a contributor to your fork, and then I can make a PR there?

**Comment by pelletier197** [comment]
Done @trevorblades :smiley:

**Comment by trevorblades** [comment]
Thanks @pelletier197! 🍣 🍔 🌮

**Pull request merged**